replication: make yield more often when recovering from long WAL (#9412)
tarantool/tarantool: changelogs/unreleased/gh-9094-recovery-from-long-WAL.md

# bugfix/replication

> * Now at replication no stop sending heartbeats during WAL scan. This prevents
>   relay from timing out when a freshly subscribed replica needs rows from the
>   end of a long `.xlog` (gh-9094).

Reviewer: ```suggestion
* Now Tarantool sends heartbeats during WAL scan when starting replication
  to a freshly subscribed replica. This prevents relay from timing out when
  the replica needs rows from the end of a long `.xlog` (gh-9094).
```
